packages/zoe/src/contractFacet/zcfMint.js

```diff
@@ -18,14 +18,14 @@ import './types.js';
 
 const { Fail } = assert;
 
-export const makeZCFMintFactory = async (
-  zcfBaggage,
-  recordIssuer,
-  getAssetKindByBrand,
-  makeEmptySeatKit,
-  reallocator,
+export const makeZCFMintFactory = (
+  /** @type {import('@agoric/vat-data').Baggage} */ zcfBaggage,
+  /** @type {{ (keyword: string, issuerRecord: IssuerRecord): void }} */ recordIssuer,
+  /** @type {GetAssetKindByBrand} */ getAssetKindByBrand,
+  /** @type { (exit?: undefined) => { zcfSeat: any; userSeat: Promise<UserSeat> }} */ makeEmptySeatKit,
+  /** @type {ZcfMintReallocator} */ reallocator,
 ) => {
-  /** @type {SetStore<ZCFMint>} The set of baggages for zcfMints */
+  /** @type {SetStore<import('@agoric/vat-data').Baggage>} The set of baggages for zcfMints */
```

Comment:
Is there any way to make this type more specific, to declare that it holds a set of `zcfMintBaggage` (which has a particular set of keys), rather than a set of generic Baggage (which itself is just a MapStore)?

Reply:
It would be pretty awkward since a MapStore allows any keys. We could have a subtype that maintains a list of keyname to type mappings and conditions `get`/`init`/`set` accordingly but I'm not sure it's a good idea and it's definitely out of scope :)